lib/modules/asset-category/AssetCategoryService.ts

```diff
@@ -202,10 +202,15 @@ export class AssetCategoryService {
       formattedValue.integer = value;
     } else if (typeof value === "boolean") {
       formattedValue.boolean = value;
-    } else if (value.lat) {
-      formattedValue.geo_point = value;
     } else if (typeof value === "object") {
-      formattedValue.object = this.formatMetadataForES(value);
+      if (
+        Object.prototype.hasOwnProperty.call(value, "lat") &&
+        Object.prototype.hasOwnProperty.call(value, "lon")
```

Comment:
You could just test existence with `value.lat !== undefiner && value.lon !== undefined`

Reply:
`hasOwnProperty` it's more safer because it test the property are directly in object not in prototype.
In most case I prefer this approach for more robust code, but in this case, it's sure just test undefined can be sufficient.

So just prefer test `typeof value.property === "undefined"`, it's more verbose but prevent re-declaration of `undefined`.

CF: https://javascript.plainenglish.io/the-different-ways-to-check-for-undefined-acaaa385a719

Reply:
Since prototype polution is forbidden as for redefining `undefined` I think the flaw described in this article does not applies to us